### ISSUE TITLE
Doesn't seem to work with my Petkit Fresh Elements Mini feeder

### ISSUE THREAD

**Issue by ydogandjiev** [jubepue/homebridge-petkit-pet-feeder#4]
**Describe the bug**
Seeing errors related to feedDailyList in the logs. My feeder seems to be found but isn't showing up in the accessories list.

**Detail Information**
1. NodeJS version:
v16.16.0

3. Homebridge version:
v1.5.0

4. Plugin version:
v3.2.0

5. Example of configure(REMOVE sensitive info!): 
```
        {
            "log_level": 2,
            "devices": [
                {
                    "headers": [
                        {
                            "key": "X-Session",
                            "value": "<SUPPRESSED>"
                        },
                        {
                            "key": "X-Api-Version",
                            "value": "8.18.1"
                        },
                        {
                            "key": "X-Timezone",
                            "value": "-7.0"
                        },
                        {
                            "key": "X-Client",
                            "value": "ios(15.5;iPhone13,3)"
                        },
                        {
                            "key": "X-TimezoneId",
                            "value": "America/Los_Angeles"
                        },
                        {
                            "key": "X-Locale",
                            "value": "en_US"
                        },
                        {
                            "key": "Accept",
                            "value": "*/*"
                        },
                        {
                            "key": "Accept-Language",
                            "value": "en-US;q=1, bg-US;q=0.9"
                        },
                        {
                            "key": "Accept-Encoding",
                            "value": "gzip, deflate"
                        },
                        {
                            "key": "Content-Type",
                            "value": "application/x-www-form-urlencoded"
                        },
                        {
                            "key": "User-Agent",
                            "value": "PETKIT/8.18.1 (iPhone; iOS 15.5; Scale/3.00)"
                        }
                    ],
                    "location": "international",
                    "model": "FeederMini",
                    "enable_http_retry": false,
                    "http_retry_count": 3,
                    "DropMeal_name": "DropMeal",
                    "MealAmount_name": "MealAmount",
                    "FeedDaily_name": "FeedDaily",
                    "FoodStorage_name": "FoodStorage",
                    "DesiccantLevel_name": "DesiccantLevel",
                    "ManualLock_name": "ManualLock",
                    "LightMode_name": "LightMode",
                    "Battery_name": "Battery",
                    "BlockDoor_name": "BlockDoor",
                    "enable_polling": true,
                    "polling_interval": 900,
                    "enable_desiccant": false,
                    "alert_desiccant_threshold": 7,
                    "enable_autoreset_desiccant": false,
                    "reset_desiccant_threshold": 5,
                    "enable_manualLock": false,
                    "enable_lightMode": false,
                    "reverse_foodStorage_indicator": false,
                    "ignore_battery_when_charging": false,
                    "fast_response": false,
                    "meal_amount": 8,
                    "enabled_daily_feeds": false,
                    "overwrite_daily_feeds": false,
                    "feed_daily_list": []
                }
            ],
            "platform": "petkit_pet_feeder"
        }
```
6. Logs(REMOVE sensitive info and homebridge debug mode):
```
[7/23/2022, 1:32:08 PM] [petkit_pet_feeder] [INFO] found you ownd one FeederMini with deviceId: 100023846.
[7/23/2022, 1:32:09 PM] [petkit_pet_feeder] [ERROR] unable to get device(100023846) status: TypeError: Cannot read properties of undefined (reading 'feedDailyList')
[7/23/2022, 1:32:09 PM] [petkit_pet_feeder] [WARN] bypass initialize Petkit Feeder device(Kitchen Feeder).
```

**Comment by jubepue** [comment]
Hi,
Please change your config to DEBUG mode and send me the results of device state:
```
[7/23/2022, 11:04:20 PM] [homebridge-petkit-pet-feeder] [DEBUG] {"result":{"btMac":"c4dd57c204fa","createdAt":"2022-06-16T21:07:08.244Z","desc":"No hay plan de alimentación.","firmware":"1.231","firmwareDetails":[{"module":"userbin","version":2228004},{"module":"lans","version":1001}],"hardware":1,"id":10014591,"locale":"Europe/Madrid","mac":"c4dd57c204f8","multiFeed":true,"multiFeedItem":{"feedDailyList":[{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":1,"suspended":1},{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":2,"suspended":1},{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":3,"suspended":1},{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":4,"suspended":1},{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":5,"suspended":1},{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":6,"suspended":1},{"items":[{"amount":40,"id":"25200","name":"Desayuno","time":25200},{"amount":30,"id":"46800","name":"Comida","time":46800},{"amount":40,"id":"72000","name":"Cena","time":72000}],"repeats":7,"suspended":1}],"isExecuted":1,"userId":"1"},"name":"Alimentador SOLO","relation":{"userId":"100350409"},"secret":"5290fd38d6111925","settings":{"desiccantNotify":1,"factor":10,"feedNotify":1,"foodNotify":1,"foodWarn":0,"foodWarnRange":[480,1200],"lightMode":0,"lightRange":[0,1440],"lowBatteryNotify":1,"manualLock":0},"shareOpen":0,"signupAt":"2022-07-22T09:09:14.935Z","sn":"23210724G10164","state":{"batteryPower":1,"batteryStatus":0,"desiccantLeftDays":10,"desiccantTime":0,"door":0,"feedState":{"addAmountTotal":60,"feedTimes":{"46800":1,"73376":1,"73417":1},"planAmountTotal":30,"realAmountTotal":90,"times":3},"feeding":0,"food":1,"ota":0,"overall":1,"pim":1,"runtime":0,"wifi":{"bssid":"c2c9e33b671d","rsq":-50,"ssid":"IOT"}},"timezone":2,"typeCode":1,"user":{"gender":1,"id":"100350409","nick":"jubepue","point":{"endGrowth":50,"growth":6,"honour":"LV0","icon":"http://img5-us.petkit.cn/misc/point/n0","icon2":"http://img5-us.petkit.cn/misc/point/l0","rank":0,"startGrowth":0}}}}
```
Tku
jubepue

**Comment by ydogandjiev** [comment]
Here you go:
```
[7/23/2022, 6:21:41 PM] [petkit_pet_feeder] Initializing petkit_pet_feeder platform...
[7/23/2022, 6:21:41 PM] [petkit_pet_feeder] [INFO] Begin to initialize Petkit feeder platform.
[7/23/2022, 6:21:41 PM] [petkit_pet_feeder] [INFO] Petkit feeder platform loaded.
[7/23/2022, 6:21:42 PM] [petkit_pet_feeder] [DEBUG] {"headers":{"X-Session":"SESSION","X-Api-Version":"8.18.1","X-Timezone":"-7.0","X-Client":"ios(15.5;iPhone13,3)","X-TimezoneId":"America/Los_Angeles","X-Locale":"en_US","Accept":"*/*","Accept-Language":"en-US;q=1, bg-US;q=0.9","Accept-Encoding":"gzip, deflate","Content-Type":"application/x-www-form-urlencoded","User-Agent":"PETKIT/8.18.1 (iPhone; iOS 15.5; Scale/3.00)"},"location":"international","model":"FeederMini","enable_http_retry":false,"http_retry_count":3,"DropMeal_name":"DropMeal","MealAmount_name":"MealAmount","FeedDaily_name":"FeedDaily","FoodStorage_name":"FoodStorage","DesiccantLevel_name":"DesiccantLevel","ManualLock_name":"ManualLock","LightMode_name":"LightMode","Battery_name":"Battery","BlockDoor_name":"BlockDoor","enable_polling":true,"polling_interval":900,"enable_desiccant":false,"alert_desiccant_threshold":7,"enable_autoreset_desiccant":false,"reset_desiccant_threshold":5,"enable_manualLock":false,"enable_lightMode":false,"reverse_foodStorage_indicator":false,"ignore_battery_when_charging":false,"fast_response":false,"meal_amount":8,"enabled_daily_feeds":0,"overwrite_daily_feeds":false,"feed_daily_list":[{"name":1,"amount":5,"time":1,"id":1}],"host":"http://api.petkt.com/latest","urls":{"owndevices":"/discovery/device_roster","deviceState":"/devicestate?id={}","deviceDetailInfo":"/device_detail?id={}","saveDailyFeed":"/save_dailyfeed?deviceId={}&day={}&time={}&amount={}","removeDailyFeed":"/remove_dailyfeed?deviceId={}&day={}&id=d{}","saveFeed":"/save_feed?deviceId={}&feedDailyList={}","dailyFeeds":"/dailyfeeds?deviceId={}&days={}","restoreDailyFeeds":"/restore_dailyfeed?deviceId={}&day={}&id=s{}","disableDailyFeeds":"/remove_dailyfeed?deviceId={}&day={}&id=s{}","resetDesiccant":"/desiccant_reset?deviceId={}","updateSettings":"/update?id={}&kv={}"},"http_options":{"method":"POST","timeout":5000,"responseType":"json","retry":{"enabled":false,"max_retry":3}},"settings":{"manualLock":"settings.manualLock","lightMode":"settings.lightMode","lightRange":"settings.lightRange","foodWarn":"settings.foodWarn","foodWarnRange":"settings.foodWarnRange"},"manufacturer":"Petkit"}
[7/23/2022, 6:21:42 PM] [petkit_pet_feeder] [INFO] initializing Petkit Feeder device.
[7/23/2022, 6:21:42 PM] [petkit_pet_feeder] [DEBUG] request device info from Petkit server.
[7/23/2022, 6:21:42 PM] [petkit_pet_feeder] [DEBUG] http://api.petkt.com/latest/discovery/device_roster
[7/23/2022, 6:21:43 PM] [petkit_pet_feeder] [DEBUG] http request success
[7/23/2022, 6:21:43 PM] [petkit_pet_feeder] [DEBUG] {"result":{"p3d":{"enable":1},"t4":{"enable":1},"r2":{"enable":1},"p3c":{"enable":1},"aqr":{"enable":1},"p3":{"enable":1},"w5c":{"enable":1},"cozy":{"gearsTotal":6,"enable":1,"tempunit":1,"sTempMin":50,"isShowDeviceSafetyClause":0,"gearsOffset":3,"sTempMax":108},"feeder":{"isShowDeviceSafetyClause":0},"h2":{"enable":0},"tools":[{"type":"walkpet","data":{}},{"type":"weight"},{"type":"remind","data":{}}],"d3":{"enable":1},"d4":{"enable":1},"w5n":{"enable":1},"aqh1":{"enable":0},"cat":{"days":30},"w5":{"enable":1},"devices":[{"data":{"name":"Kitchen Feeder","createdAt":"2020-08-23T21:21:10.028Z","id":100023846,"state":1,"dailyFeed":{"amount":210,"day":20220723,"deviceId":0,"items":[],"realAmount":265},"relation":{"petIds":["100158096","100158097","100158098"],"userId":"100114864"},"status":{"batteryPower":4,"batteryStatus":0,"desiccantLeftDays":23,"desiccantTime":0,"errorPriority":0,"feeding":0,"food":1,"ota":0,"overall":1,"pim":1,"runtime":80128,"wifi":{"bssid":"605f8dc5bc86","rsq":-51,"ssid":"Boom Boom"}},"desc":"Next feeding time: Sun. 00:00"},"type":"FeederMini"}],"k2":{"enable":1},"go":{"hasMarks":0,"maxMarksPerRoute":30,"markEnabled":1},"k3":{"enable":1},"aq":{"enable":1},"hasRelay":false,"feedermini":{"isShowDeviceSafetyClause":0},"w4x":{"enable":1},"appbanner":[],"t3":{"enable":1},"aq1s":{"enable":1}}}
[7/23/2022, 6:21:43 PM] [petkit_pet_feeder] [INFO] found you ownd one FeederMini with deviceId: 100023846.
[7/23/2022, 6:21:43 PM] [petkit_pet_feeder] [DEBUG] request initial device status from Petkit server.
[7/23/2022, 6:21:43 PM] [petkit_pet_feeder] [DEBUG] http://api.petkt.com/latest/feedermini/device_detail?id=100023846
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [DEBUG] http request success
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [DEBUG] {"result":{"createdAt":"2020-08-23T21:21:10.028Z","feed":{"deviceId":100023846,"isExecuted":1,"isLock":1,"items":[{"amount":15,"id":"0","name":"Night snack","petAmount":[{"amount":5,"petId":"100158098"},{"amount":5,"petId":"100158096"},{"amount":5,"petId":"100158097"}],"time":0},{"amount":45,"id":"25200","name":"Breakfast","petAmount":[{"amount":20,"petId":"100158098"},{"amount":25,"petId":"100158096"}],"time":25200},{"amount":25,"id":"25320","name":"Breakfast","petAmount":[{"amount":25,"petId":"100158097"}],"time":25320},{"amount":40,"id":"43200","name":"Lunch","petAmount":[{"amount":20,"petId":"100158098"},{"amount":20,"petId":"100158096"}],"time":43200},{"amount":25,"id":"43320","name":"Lunch","petAmount":[{"amount":25,"petId":"100158097"}],"time":43320},{"amount":40,"id":"64800","name":"Dinner","petAmount":[{"amount":20,"petId":"100158098"},{"amount":20,"petId":"100158096"}],"time":64800},{"amount":20,"id":"64920","name":"Dinner","petAmount":[{"amount":20,"petId":"100158097"}],"time":64920}],"repeats":"1,2,3,4,5,6,7","suspended":0,"userId":"100114864"},"firmware":"1.394","firmwareDetails":[{"module":"userbin","version":2003001}],"hardware":1,"id":100023846,"locale":"America/Los_Angeles","mac":"dabfc002fd2e","name":"Kitchen Feeder","relation":{"petIds":["100158096","100158097","100158098"],"userId":"100114864"},"secret":"9a26fd4ff263e04d","settings":{"desiccantNotify":0,"feedNotify":0,"foodNotify":1,"lightMode":1,"lightRange":[0,1440],"manualLock":0,"reBatteryNotify":1},"shareOpen":1,"signupAt":"2022-07-23T02:11:02.948Z","sn":"55200613B80144","state":{"batteryPower":4,"batteryStatus":0,"desiccantLeftDays":23,"desiccantTime":0,"errorPriority":0,"feeding":0,"food":1,"ota":0,"overall":1,"pim":1,"runtime":80128,"wifi":{"bssid":"605f8dc5bc86","rsq":-51,"ssid":"Boom Boom"}},"timezone":-8,"type":0,"user":{"avatar":"http://img5-us.petkit.cn/uavatar/2020/8/24/5f42dd7befad9b2b463a0511VaGmUDOab","gender":1,"id":"100114864","nick":"ydogandjiev","point":{"endGrowth":2000,"growth":1421,"honour":"LV5","icon":"http://img5-us.petkit.cn/misc/point/n5","icon2":"http://img5-us.petkit.cn/misc/point/l5","rank":5,"startGrowth":1000}},"userId":"100114864"}}
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [ERROR] unable to get device(100023846) status: TypeError: Cannot read properties of undefined (reading 'feedDailyList')
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [WARN] bypass initialize Petkit Feeder device(Kitchen Feeder)., 6:21:43 PM]
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [DEBUG] http request success
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [DEBUG] {"result":{"createdAt":"2020-08-23T21:21:10.028Z","feed":{"deviceId":100023846,"isExecuted":1,"isLock":1,"items":[{"amount":15,"id":"0","name":"Night snack","petAmount":[{"amount":5,"petId":"100158098"},{"amount":5,"petId":"100158096"},{"amount":5,"petId":"100158097"}],"time":0},{"amount":45,"id":"25200","name":"Breakfast","petAmount":[{"amount":20,"petId":"100158098"},{"amount":25,"petId":"100158096"}],"time":25200},{"amount":25,"id":"25320","name":"Breakfast","petAmount":[{"amount":25,"petId":"100158097"}],"time":25320},{"amount":40,"id":"43200","name":"Lunch","petAmount":[{"amount":20,"petId":"100158098"},{"amount":20,"petId":"100158096"}],"time":43200},{"amount":25,"id":"43320","name":"Lunch","petAmount":[{"amount":25,"petId":"100158097"}],"time":43320},{"amount":40,"id":"64800","name":"Dinner","petAmount":[{"amount":20,"petId":"100158098"},{"amount":20,"petId":"100158096"}],"time":64800},{"amount":20,"id":"64920","name":"Dinner","petAmount":[{"amount":20,"petId":"100158097"}],"time":64920}],"repeats":"1,2,3,4,5,6,7","suspended":0,"userId":"100114864"},"firmware":"1.394","firmwareDetails":[{"module":"userbin","version":2003001}],"hardware":1,"id":100023846,"locale":"America/Los_Angeles","mac":"dabfc002fd2e","name":"Kitchen Feeder","relation":{"petIds":["100158096","100158097","100158098"],"userId":"100114864"},"secret":"9a26fd4ff263e04d","settings":{"desiccantNotify":0,"feedNotify":0,"foodNotify":1,"lightMode":1,"lightRange":[0,1440],"manualLock":0,"reBatteryNotify":1},"shareOpen":1,"signupAt":"2022-07-23T02:11:02.948Z","sn":"55200613B80144","state":{"batteryPower":4,"batteryStatus":0,"desiccantLeftDays":23,"desiccantTime":0,"errorPriority":0,"feeding":0,"food":1,"ota":0,"overall":1,"pim":1,"runtime":80128,"wifi":{"bssid":"605f8dc5bc86","rsq":-51,"ssid":"Boom Boom"}},"timezone":-8,"type":0,"user":{"avatar":"http://img5-us.petkit.cn/uavatar/2020/8/24/5f42dd7befad9b2b463a0511VaGmUDOab","gender":1,"id":"100114864","nick":"ydogandjiev","point":{"endGrowth":2000,"growth":1421,"honour":"LV5","icon":"http://img5-us.petkit.cn/misc/point/n5","icon2":"http://img5-us.petkit.cn/misc/point/l5","rank":5,"startGrowth":1000}},"userId":"100114864"}}
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [ERROR] unable to get device(100023846) status: TypeError: Cannot read properties of undefined (reading 'feedDailyList')
[7/23/2022, 6:21:45 PM] [petkit_pet_feeder] [WARN] bypass initialize Petkit Feeder device(Kitchen Feeder).
```

**Comment by jubepue** [comment]
Hi,
A new release is uploaded, please test it and tell me back.
https://github.com/jubepue/homebridge-petkit-pet-feeder/releases/tag/v3.2.1

Regards
jubepue

**Comment by ydogandjiev** [comment]
Thank you for the quick turnaround! Works great with my feeder after the latest update.

**Comment by jubepue** [comment]
Hi Yuri,Do you have any problem with the dailyFeeds plan? Because other user have it with an feeder mini proEnviado desde mi iPhoneEl 26 jul 2022, a las 23:01, Yuri Dogandjiev ***@***.***> escribió:﻿
Closed #4 as completed.
